include/sys/zfs_context.h

```diff
@@ -32,7 +32,7 @@
 extern "C" {
 #endif
 
-#ifdef __KERNEL__
```

Comment:
This would be a good place to document exactly what the user, `__KERNEL__` , and `_STANDALONE` contexts are.

Reply:
Great idea. They are so second nature to me I forget that they aren't to others and a little context can go a long way.

Reply:
done.